feat: webpack combines all js and scss into one js

FinnYguY opened the pull request:
Index.js file adds div into body, gives it "appendDiv" id and calls for "addText" function from writer.js which adds chosen text to selected element.
There is one styles.scss file for page styling.
All js and scss files are combined by webpack into one main.js file, which is added in a <script> tag to index.html

FinnYguY committed:
feat: webpack combines all js and scss into one js

avkulahina commented:
how can I check 16k files? is it all your job?
![Screenshot from 2020-06-15 13-57-56](https://user-images.githubusercontent.com/56341537/84650242-85f53a80-af10-11ea-8519-e0d2e81a7270.png)

joking. the first thing you need to do - remove "generated" files from your project and then create .gitignore file, which will exclude these files from working directory. You don't need:

- node_modules/
- package.json **Sorry, I meant package-lock.json instead**
- dist/main.js

Please do changes, and revert back to me.

avkulahina commented:
Could you add me as collaborator (avkulahina) so I can comment each line in the future.

Also please, I want to check the rendered job as well ( site). Could you please deploy it? You can use GitHub Pages feature for example. [Only thing you need is to create and push gh-pages branch](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).

FinnYguY committed:
fix: ignored all unwanted files

FinnYguY committed:
Update .gitignore

FinnYguY committed:
Delete main.js

FinnYguY committed:
Update .gitignore

FinnYguY committed:
test for adding main.js

FinnYguY commented:
> Could you add me as collaborator (avkulahina) so I can comment each line in the future.
> 
> Also please, I want to check the rendered job as well ( site). Could you please deploy it? You can use GitHub Pages feature for example. [Only thing you need is to create and push gh-pages branch](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).

FinnYguY commented:
Missclicked reopening the pull request. I .gitignored almost all the unnecessary files, except main.js (For I believe it is needed for my index.html). I sent u request to become a collaborator and deployed the test result https://finnyguy.github.io/memory-game/

avkulahina commented:
> Missclicked reopening the pull request. I .gitignored almost all the unnecessary files, except main.js (For I believe it is needed for my index.html). I sent u request to become a collaborator and deployed the test result https://finnyguy.github.io/memory-game/

you are right. you need main.js, it is not need in project, but need for your gh-pages branch, so better you keep it.

avkulahina commented on .gitignore:
```diff
@@ -0,0 +1,7 @@
+node_modules/
+
+#package config files
+package.json
+package-lock.json
+webpack.config.json
```
please, explude package.json and webpack.config.json from .gitignore. It was my mistake I told you package.json is not needed. I was thinking about package-lock.json.

Also keep in mind there always a need of an empty line in the end of the file

avkulahina commented on memory-game-project/src/styles.scss:
```diff
@@ -0,0 +1,16 @@
+  justify-content: center;
+  align-items: center;
+}
+#appendDiv {
+  background-color: $green;
+  padding: 20px;
+  margin: 20px;
+  border: 2px solid black;
```
avoid using id in selectors. It is always better to use class instead. also scss give you an ability to *connect* styles from the same block. e.g. following is equals to your definition:
```
body {
  font-family: $font;
  display: flex;
  justify-content: center;
  align-items: center;
  
  #appendDiv {
    background-color: $green;
    padding: 20px;
  }
}
```

so when you are creating styles it is better to understand the structure, as for me.

also scss is very helpful when using [BEM](http://getbem.com/naming/)

e.g.
```
.lovely-header {
  &__button {
    &--active {
    }
  }
}
```
will generate 3 styles:
```
.lovely-header {
 }

.lovely-header__button {
}

.lovely-header__button--active {
}
```

avkulahina commented on memory-game-project/webpack.config.js:
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.s[ac]ss$/i,
```
one more reccomendation: try to use webpack-dev-server. It is very comfortable for developing because it compile code automatically and update the page in browser. 
https://webpack.js.org/guides/development/#using-source-maps
https://webpack.js.org/guides/development/#using-webpack-dev-server

avkulahina reviewed (changes requested):
pls do changes

FinnYguY committed:
added package and webpack.config

FinnYguY committed:
reworked scss and gitignore

FinnYguY committed:
added webpack-dev-server support

FinnYguY commented:
So I added the package.json and webpack.config.js files, reworked scss and js for it to work with classes not id's and connected the style rules in scss. At last I added the webpack-dev-server support

avkulahina commented:
LGFM (looks good for me). You can now "Merge pull request".
I will prepare next task for you
@FinnYguY
